mage/query-modules/python/gnn-node-classification.md

```diff
@@ -0,0 +1,343 @@
+| path_to_model              | String       | "/tmp/torch_models"                                                        | Path from where model is loaded and where is stored                                                                                                                                   |
+
+#### Exceptions:
+- `Exception`: exception is raised if some variable in dictionary params is not correctly defined.
+
+#### Output
+- `mgp.Record(
+    hidden_features_size=list,
+    layer_type=str,
+    aggregator=str,
+    learning_rate=float,
+    weight_decay=float,
+    split_ratio=float,
+    metrics=mgp.Any,
+    node_id_property=str,
+    num_epochs=int,
+    console_log_freq=int,
+    checkpoint_freq=int,
+    device_type=str,
+    path_to_model=str,
+)` ➡ Map of parameters set for training
+
+#### Usage:
+
+```cypher
+  CALL node_classification.set_model_parameters(
+    {layer_type: "GATJK", learning_rate: 0.001, hidden_features_size: [16,16], class_name: "fraud", features_name: "embedding"}
+  ) YIELD * RETURN *;
+```
+
+### `train(num_epochs)`
+
+This procedure performs model training. Firstly it declares data, model, optimizer and criterion. Afterwards it performs training.
+#### Input
+- `num_epochs (int, optional)` ➡ number of epochs. Default set to 100.
+
+#### Exceptions
+- `Exception`➡ raised if graph is empty
+
+#### Outputs
+- `epoch: int` ➡ epoch number
+- `loss: float`➡ loss of model on training data
+- `val_loss: float`➡ loss of model on validation data
+- `train_log: list`➡ list of metrics on training data
+- `val_log: list`➡ list of metrics on validation data
+
+#### Usage
+```cypher
+  CALL node_classification.train() YIELD * RETURN *;
+```
+
+### `get_training_data()`
+Use following procedure to get logged data from training.
+
+#### Return values
+- `epoch: int` ➡ epoch number for current record's logged data
+- `loss: float`➡ loss in epoch
+- `train_log: mgp.Any` ➡ training parameters for epoch
+- `val_log: mgp.Any`➡ validation parameters for epoch
+
+#### Usage
+```cypher
+  CALL node_classification.get_training_data() YIELD * RETURN *;
+```
+
+### `save_model()`
+
+This function saves model to model saving folder. If there are already total of **max_models_to_keep** models in model saving folder, 
+the oldest model is deleted.
+
+#### Exception
+- `Exception`: raised if model is not initialized or defined
+
+#### Return values
+- `path (str)`➡ path to stored model
+- `status (str)`➡ status of stored model
+
+#### Usage
+```cypher
+  CALL node_classification.save_model() YIELD * RETURN *;
+```
+
+### `load_model(num)`
+
+This function loads model from specified folder.
+
+#### Input
+
+- `num (int, optional)`: ordinary number of model to load from default path on disc. Defaults to 0 (newest model).
```

Comment:
```suggestion
- `num (int, optional)`: ordinary number of model to load from the default path on the disc (default: 0, i.e., newest model).
```

Reply:
ordinary number? what do you mean?

Reply:
It’s got to be ordinal number

Reply:
However, this seems like bad UX to me and I believe we need to do it differently @antoniofilipovic

Reply:
I agree completely :) we can add task to MAGE to fix this  bug